lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,393 @@
+      },
+    );
+  }
+
+  /**
+   * This calculates branded tokens equivalent to given value tokens.
+   *
+   * @param valueTokens Amount of value token.
+   *
+   * @return {Promise<string>} Promise that resolves to amount of branded token.
+   */
+  convertToBrandedTokens(valueTokens) {
+    return this.contract.methods
+      .convertToBrandedTokens(valueTokens)
+      .call();
+  }
+
+  /**
+   * Request stake for given amount. Approval for stake amount to branded
+   * token is required before calling this method.
+   *
+   * @param stakeAmount Stake amount.
+   * @param txOptions Transaction options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async requestStake(stakeAmount, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(
+        `Invalid from address ${txOptions.from} in transaction options.`,
+      );
+      return Promise.reject(err);
+    }
+
+    const mintedAmount = await this.convertToBrandedTokens(stakeAmount);
+    const tx = await this.requestStakeRawTx(stakeAmount, mintedAmount);
+    return Utils.sendTransaction(tx, txOptions);
+  }
+
+  /**
+   * Raw tx for request stake.
+   *
+   * @param stakeAmount Stake amount.
+   * @param mintAmount Amount that will be minted after staking.
+   *
+   * @return Promise<Object> Raw transaction object.
+   */
+  requestStakeRawTx(stakeAmount, mintAmount) {
+    return Promise.resolve(this.contract.methods.requestStake(stakeAmount, mintAmount));
+  }
+
+  /**
+   * Accept open stake request identified by request hash.
+   *
+   * @param stakeRequestHash Stake request hash.
+   * @param r R of signature received from worker.
+   * @param s s of signature received from worker.
+   * @param v v of signature received from worker.
+   * @param txOptions Transaction options
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async acceptStakeRequest(stakeRequestHash, r, s, v, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(
+        `Invalid from address ${txOptions.from} in transaction options.`,
+      );
+      return Promise.reject(err);
+    }
+
+    const tx = await this.acceptStakeRequestRawTx(stakeRequestHash, r, s, v);
+    return Utils.sendTransaction(tx, txOptions);
+  }
+
+  /**
+   * Raw transaction for accept stake request.
+   *
+   * @param stakeRequestHash Stake request hash.
+   * @param r R of signature received from worker.
+   * @param s s of signature received from worker.
+   * @param v v of signature received from worker.
+   *
+   * @return Promise<Object> Raw transaction object.
```

Comment:
```suggestion
   * @return {Promise<Object>} Raw transaction object.
```
Other places as well